lib/kafkajs/_consumer.js

```diff
@@ -1336,22 +1336,30 @@ class Consumer {
    */
   async commitOffsets(topicPartitions = null) {
     if (this.#state !== ConsumerState.CONNECTED) {
-      throw new error.KafkaJSError('Commit can only be called while connected.', { code: error.ErrorCodes.ERR__STATE });
+      return Promise.reject(new error.KafkaJSError('Commit can only be called while connected.', { code: error.ErrorCodes.ERR__STATE }));
     }
 
-    try {
-      if (topicPartitions === null) {
-        this.#internalClient.commitSync();
-      } else {
-        const topicPartitionsRdKafka = topicPartitions.map(
-          topicPartitionOffsetMetadataToRdKafka);
-        this.#internalClient.commitSync(topicPartitionsRdKafka);
-      }
-    } catch (e) {
-      if (!e.code || e.code !== error.ErrorCodes.ERR__NO_OFFSET) {
-        throw createKafkaJsErrorFromLibRdKafkaError(e);
+    return new Promise((resolve, reject) => {
+      try {
+        let cb = (e) => {
+          if (e)
+            reject(createKafkaJsErrorFromLibRdKafkaError(e));
+          else
+            resolve();
+        };
+
+        if (topicPartitions)
+          topicPartitions = topicPartitions.map(topicPartitionOffsetToRdKafka);
```

Comment:
```suggestion
          topicPartitions = topicPartitions.map(topicPartitionOffsetMetadataToRdKafka);
```

This will make the failing test (commit.spec.js) pass